compiler/semstmts.nim

```diff
@@ -1639,6 +1651,20 @@ proc addResult(c: PContext, n: PNode, t: PType, owner: TSymKind) =
       n.add newSymNode(c.p.resultSym)
     addParamOrResult(c, c.p.resultSym, owner)
 
+proc addDefaultFieldForResult(c: PContext, n: PNode, t: PType) =
```

Comment:
Why?

Reply:
`addDefaultFieldForResult` adds default values for implicit variables within procs:

transform
```nim
proc foo(): Object =
  discard
```
into
```nim
proc foo(): Object =
  result = Object()
  discard
```
So that default values are applied.

Reply:
This is unnecessary if we do https://github.com/nim-lang/RFCs/issues/378 and I think we should.

Reply:
Sure.

Reply:
I will undo the default fields for results.